lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -2774,11 +2781,20 @@ constructTransaction api argGenChange knownPools poolStatus apiWalletId body = d
                     db currentEpochSlotting knownPools
                     poolStatus withdrawal
 
+        optionalVoteAction <- case (body ^. #vote) of
+            Just (ApiT action) ->
+                liftIO $ Just <$> WD.voteAction trWorker db action
+            Nothing ->
+                liftIO $ W.handleVotingWhenMissingInConway nl db
+                              (isJust (body ^. #delegations))
```

Comment:
When it comes to constructTransaction endpoint this is part of NEW transaction flow where we have construct/sign/decode/submit that contains all new features that were are in the last two years. joinStakePool like postTransaction are legacy endpoints (they have construct/sign/submit in one step). Of course I understand that we might ALSO support joinStakePool having conway voting feature as daedalus might use it (although not sure if they have not adopted new tx workflow). Nevertheless, I do think voting should be primarly supported in constructTransaction....

Reply:
> Nevertheless, I do think voting should be primarly supported in constructTransaction....

Yes, I do agree with that — what I'm saying is that in `constructTransaction`, voting should be explicit rather than automatic/implicit. In other words, `constructTransaction` should not implicitly vote `Abstain` if the JSON data does not explicitly contain a vote to `abstain`.

> Of course I understand that we might ALSO support joinStakePool having conway voting feature as daedalus might use it (although not sure if they have not adopted new tx workflow).

Yes. I believe that Daedalus is still using `joinStakePool`, but worth checking.